addons/spreadsheet_account/models/account.py

```diff
@@ -58,16 +58,22 @@ def _build_spreadsheet_formula_domain(self, formula_params):
             ("date", ">=", start),
             ("date", "<=", end),
         ]
+        # It is more optimized to (like) search for code directly in account.account than in account_move_line
         code_domain = expression.OR(
             [
-                expression.AND([
-                    [("account_id.code", "=like", f"{code}%")],
-                    expression.OR([balance_domain, pnl_domain]),
-                ])
-                for code in codes
+                ("code", "=like", f"{code}%"),
             ]
+            for code in codes
         )
-        domain = expression.AND([code_domain, [("company_id", "=", company_id)]])
+        full_codes = self.env["account.account"].search(code_domain).read(["code"])
+        code_domain = expression.OR(
+            [
+                ("account_id.code", "=", f"{code_dict['code']}"),
+            ]
+            for code_dict in full_codes
+        )
```

Comment:
It could also be an "in" operator. Looks simpler in the code. I'm not sure how/if it would affect the query's performance. Could you check ?
```suggestion
        code_domain = [("account_id.code", "in", full_codes)]
```

Reply:
I don't think it will but I'll check

Reply:
I don't see any impact